packages/client/Usage.md

```diff
@@ -1,23 +1,134 @@
-# TypeScript SDK
+# Xata SDK Reference
 
-There're four types of objects in the Xata TypeScript SDK:
+There are four types of objects in the Xata TypeScript SDK:
 
-- `Repository`: a table representation that can be used to create, read, update, and delete records.
 - `Query`: a combination of filters and other parameters to retrieve a collection of records.
+- `Repository`: a table representation that can be used to create, read, update, and delete records.
 - `XataRecord`: a row in a table.
 - `Page`: a collection of records that can be paginated.
 
-## Repository
+Let's explore them below.
+
+## Table of Contents
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Query](#query)
+  - [Column selection](#column-selection)
+  - [Sorting](#sorting)
+  - [Filtering](#filtering)
+  - [Combining queries](#combining-queries)
+- [Repository](#repository)
+  - [Reading records](#reading-records)
+  - [Creating records](#creating-records)
+  - [Updating records](#updating-records)
+  - [Deleting records](#deleting-records)
+  - [Searching records](#searching-records)
+- [Page](#page)
+  - [Iterators and generators](#iterators-and-generators)
+  - [Helper variables](#helper-variables)
+- [XataRecord](#xatarecord)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Query
+
+To get a collection of records, you can use the `Query` object. It provides the following methods:
+
+- `getFirst()`: returns the first record in the query results.
+- `getPaginated()`: returns a page of records in the query results.
+- `getAll()`: returns all the records in the query results.
```

Comment:
Not discussing the method name here, but there's a threshold for the `getAll` method.

Though conflicting with the method's name, I think it's best to be explicit about it in the docs to avoid confusion and issues for consumers.

```suggestion
- `getAll()`: returns the first 200 records in the query results.
```

Reply:
This is actually a really good point to address. I thought get all actually does get all, by getting 200 and then appending the next 200 and the next and the next until all records are available. 

I haven’t yet looked into the implementation so I’m not sure if that happens or if it gets only 200, so thanks! Now I can go dive into the implementation and find out!

Reply:
That's not what happens 😅

`getAll` gets all the records as I wrote down. There's currently no method that returns 200 also. `getMany` returns 20 which is the default pagination size. And even in that case I wouldn't write the number in the docs, as there's a point of the helper variables that keep track of these "hardcoded" values.